### ISSUE TITLE
Refactor the Install/Uninstall methods of the installer

### ISSUE THREAD

**Issue by jamiemonserrate** [vmware-tanzu/cluster-api-provider-bringyourownhost#221]
**Describe the solution you'd like**
The Install/Uninstall methods currently modify the state of the installer - which is a bad design. See discussion [here](https://github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/pull/216#discussion_r744643054)

This code was written when we assumed that we would have the bundleRepo Address at startup time. This assumption is not true, and we will only know this information at Install/Uninstall time. 

So let's refactor the installer to reflect this updated assumption.

**Comment by jeuxdemains** [comment]
Me and Alex will take this one.

**Comment by jeuxdemains** [comment]
Here is an update on the task after gathering info about what we need to do:

- during instantiation of the Installer, it should not require or depend on the Bundle Registry which is currently passed to the installer.New() method at creation time (see New() in installer.go, the argument is called bundleRepo)
- once instantiated, the Installer should receive the Bundle Registry (bundleRepo) via setter so it could be set and/or updated anytime
 
The final goal is to have the Installer instance created just once during the lifetime of the installation process and manage its Bundle Registry on-the-fly.
This will also enable performing basic checks (supported OS, pre-requisite packages integrity, etc.) at earlier stage of the installation procedure, at point when the Bundle Registry is not yet known.

**Comment by jamiemonserrate** [comment]
> once instantiated, the Installer should receive the Bundle Registry (bundleRepo) via setter so it could be set and/or updated anytime

I don't think this should be via a setter. I think this should be part of the `Install` contract itself. Otherwise the caller will have to remember to call the setter before calling `Install`, which is confusing and error prone.

**Comment by jeuxdemains** [comment]
You're right. As part of the contract between the user and the Installer is a better solution. Will be updated. 

Download path will be updated as well to avoid clashes.
